src/object/cli_ec.c

```diff
@@ -346,6 +346,7 @@ obj_ec_recx_scan(daos_iod_t *iod, d_sg_list_t *sgl,
 				return rc;
 			ec_recx = ec_recx_array->oer_recxs;
 		}
+		D_ASSERT(ec_recx != NULL);
```

Comment:
yes, looks it supposes that 'ec_recx_array->oer_recxs' is NULL at line 343, but I think this assert should be enough, it can also catch exceptions after ob_ec_recxs_init() call above.

Reply:
the handling is per recx, if iod->iod_nr > 1 then it is possible ec_recx_array->oer_recxs is non-NULL at line 343, for example there is full-stripe in 2nd recx and inited it, then at 3rd recx it is non-NULL.